### PR TITLE
feat(arriba): recipe

### PR DESCRIPTION
**Pull request by jemten**
### This PR fixes:

- see above

### How to test:

- Automatic and continuous test pass

### Expected outcome:
- Installation, unit and integration tests pass

### Review:
- [ ] Code review
- [ ] New code is executed and covered by tests
- [ ] Tests pass

### CODE REVIEW

File: lib/MIP/Check/Parameter.pm [hunk 1]
```diff
@@ -1962,9 +1963,12 @@ sub check_salmon_compatibility {
     use MIP::Set::Parameter qw{ set_recipe_mode };
 
     ## Check if program is gong to run
```
Comment (henrikstranneheim) [inline lib/MIP/Check/Parameter.pm]:
```suggestion
    ## Check if program is going to run
```

File: lib/MIP/Check/Parameter.pm [hunk 2]
```diff
@@ -1980,30 +1984,30 @@ sub check_salmon_compatibility {
 
         ## Turn of Salmon if multiple sequence types are present
```
Comment (henrikstranneheim) [inline lib/MIP/Check/Parameter.pm]:
```suggestion
        ## Turn of recipe if multiple sequence types are present
```

File: lib/MIP/Check/Parameter.pm [hunk 3]
```diff
@@ -1980,30 +1984,30 @@ sub check_salmon_compatibility {
...
         if ( uniq( values %sequence_run_type ) > 1 ) {
-            $is_salmon_compatible = 0;
+            $is_compatible = 0;
         }
     }
 
-    if ( not $is_salmon_compatible ) {
+    if ( not $is_compatible ) {
 
         ## Turn of salmon and downstream recipes
```
Comment (henrikstranneheim) [inline lib/MIP/Check/Parameter.pm]:
```suggestion
        ## Turn of current recipe and downstream recipes
```

File: lib/MIP/Check/Pipeline.pm [hunk 4]
```diff
@@ -1172,16 +1171,22 @@ sub check_rd_rna {
...
+
+  RECIPE:
+    foreach my $recipe (@recipes_to_check) {
+
+        check_recipe_fastq_compatibility(
+            {
+                active_parameter_href   => $active_parameter_href,
+                infile_lane_prefix_href => $infile_lane_prefix_href,
+                parameter_href          => $parameter_href,
+                recipe_name             => q{salmon_quant},
```
Comment (henrikstranneheim) [inline lib/MIP/Check/Pipeline.pm]:
```suggestion
                recipe_name             => $recipe,
```

File: lib/MIP/Get/Executable.pm [hunk 5]
```diff
@@ -64,6 +64,7 @@ sub get_binary_version {
   BINARY:
     while ( my ( $binary, $binary_path ) = each %{$binary_info_href} ) {
 
+        say STDERR $binary;
```
Comment (henrikstranneheim) [inline lib/MIP/Get/Executable.pm]: leftover from debugging perhaps?
```suggestion
```

File: lib/MIP/Recipes/Analysis/Arriba.pm [hunk 6]
```diff
@@ -0,0 +1,438 @@
...
+        }
+      );
+    push @arriba_commands, $PIPE;
+
+    push @arriba_commands,
+      arriba(
+        {
+            infile_path                => q{/dev/stdin},
+            outfile_path               => $outfile_path,
+            discarded_fusion_file_path => $outfile_path_prefix
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Analysis/Arriba.pm]: Order AB

File: lib/MIP/Recipes/Analysis/Arriba.pm [hunk 7]
```diff
@@ -0,0 +1,438 @@
...
+        {
+            outfile_paths_ref => [$star_outfile_path],
+        }
+      );
+    push @arriba_commands, $PIPE;
+
+    push @arriba_commands,
+      arriba(
+        {
+            infile_path                => q{/dev/stdin},
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Analysis/Arriba.pm]:
```suggestion
            infile_path                => catfile( dirname(devnull), q{stdin}),
```
From: File::Spec->devnull)

File: lib/MIP/Set/Parameter.pm [hunk 8]
```diff
@@ -1034,6 +1027,9 @@ sub set_recipe_mode {
 
     check( $tmpl, $arg_href, 1 ) or croak q{Could not parse arguments!};
 
+    ## Retrieve logger object
+    my $log = Log::Log4perl->get_logger($LOG_NAME);
```
Comment (henrikstranneheim) [inline lib/MIP/Set/Parameter.pm]: 👍

File: lib/MIP/Recipes/Analysis/Arriba.pm [hunk 9]
```diff
@@ -0,0 +1,440 @@
...
+
+            ## Add read two file index array
+            push @reverse_files, $infile_paths[$paired_end_tracker];
+        }
+
+        ## Increment paired end tracker
+        $paired_end_tracker++;
+    }
+
+    my @fastq_files =
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Analysis/Arriba.pm]: How would this work for a combination of single or paired-end reads?

Comment (jemten) [reply lib/MIP/Recipes/Analysis/Arriba.pm]: It doesn't. That's why we run a check earlier and turn off arriba if it is a mix of SE and PE

Comment (henrikstranneheim) [reply lib/MIP/Recipes/Analysis/Arriba.pm]: 👍

**Pull request merged**